app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, useEffect, useState, useRef } from 'react';
+import { FormControl } from '@material-ui/core';
+import T from 'i18n-react';
+import {
+  ICreateAndEditRecipeFormProps,
+  IRecipeData,
+} from 'components/CreateAndEditRecipeForm/types';
+import {
+  FormFieldWrapper,
+  NormalLabel,
+  StyledForm,
+  StyledRecipeNameTextField,
+  StyledDescriptionTextArea,
+  StyledCancelButton,
+  StyledSaveButton,
+  StyledFormButtonWrapper,
+  getLabelStyle,
+} from 'components/CreateAndEditRecipeForm/styledComponents';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+import { debounce } from 'lodash';
+
+export default function({
+  recipeData,
+  setIsCreateAndEditRecipeFormOpen,
+  setSnackbar,
+  recipeFormAction,
```

Comment:
i see only 1 recipeFormAction i.e. CreateRecipe
plz let me know if we have any more actions that will be triggered from this form
if not then can we update this code to reduce if any ambiguity

plz provide proper explanation for yes and no either way

Reply:
yes we need this action type as we are using the same form component create and edit